pwnlib/tubes/process.py

```diff
@@ -892,14 +927,92 @@ def libc(self):
         ELF('/lib64/libc-...so')
         >>> p.close()
         """
+        lib = self._waitfor_libc(timeout=0)
+        lib.describe()
+        return lib
+
+    def _libc(self):
+        """Function to retrieve the libc library from the current process
+
+        Returns:
+            ELF or None
+        """
+
         from pwnlib.elf import ELF
 
         for lib, address in self.libs().items():
             if 'libc.so' in lib or 'libc-' in lib:
-                e = ELF(lib)
+                e = ELF(lib, checksec=False)
                 e.address = address
                 return e
 
+    def _waitfor_libc(self, timeout=1):
+        """Function to retrieve the libc library from the current process waiting for
+        it to be loaded
+
+        Arguments:
+            timeout(int): Time to wait for libc to being loaded
+
+        Raises:
+            PwnlibException: In case the libc is not found in the given timeout
+
+        Returns:
+            ELF
+        """
+
+        limit = time.time() + timeout
```

Comment:
Use a `Timeout` object instead of manually implementing a timeout

Reply:
Done in edeca4f